src/org/opensolaris/opengrok/web/Util.java

```diff
@@ -1840,4 +1842,45 @@ public static String completeUrl(String url, HttpServletRequest req) {
             return url;
         }
     }
+
+    /**
+     * Parses the specified url and returns its query params.
+     * @param url url to retrieve the query params from
+     * @return query params of {@code url}
+     */
+    public static Map<String, List<String>> getQueryParams(final URL url) {
+        if (url == null) {
+            throw new IllegalArgumentException("Cannot get query params from the null url");
+        }
+        Map<String, List<String>> returnValue = new HashMap<>();
+
+        if (url.getQuery() == null) {
+            return returnValue;
+        }
+
+        String[] pairs = url.getQuery().split("&");
+
+        for (String pair : pairs) {
+            if (pair.isEmpty()) {
+                continue;
+            }
+
+            int idx = pair.indexOf('=');
```

Comment:
I will create a test case for that :)

Reply:
I thought that param without `=` is not valid but here they say otherwise <https://stackoverflow.com/questions/4557387/is-a-url-query-parameter-valid-if-it-has-no-value>

So I repaired it, good catch :)